tests/unit/handle_trailer_test.sh

```diff
@@ -0,0 +1,260 @@
+Reviewed-by: John Doe <johndoe@community.com>
+
+Signed-off-by: kw <kw@kw>
+Reviewed-by: John Doe <johndoe@community.com>
+
+Signed-off-by: kw <kw@kw>
+Reviewed-by: John Doe <johndoe@community.com>
+
+Signed-off-by: kw <kw@kw>"
+
+# Correct trailers for --add-acked-by and -a
+CORRECT_ACKED_COMMIT="Signed-off-by: kw <kw@kw>
+Acked-by: Michael Doe <michaeldoe@community.com>
+
+Signed-off-by: kw <kw@kw>"
+
+CORRECT_ACKED_LOG="Signed-off-by: kw <kw@kw>
+Acked-by: Michael Doe <michaeldoe@community.com>
+
+Signed-off-by: kw <kw@kw>
+Acked-by: Michael Doe <michaeldoe@community.com>
+
+Signed-off-by: kw <kw@kw>
+Acked-by: Michael Doe <michaeldoe@community.com>
+
+Signed-off-by: kw <kw@kw>"
+
+# Correct trailers for --add-fixes and -f
+# Appending the correct hash is needed. This has
+# to be done during test runs, since hashes
+# are randomly generated.
+CORRECT_FIXED_COMMIT="Signed-off-by: kw <kw@kw>
+Fixes: <hash>
+
+Signed-off-by: kw <kw@kw>"
+
+# Correct trailers when using adding Acked-by, then
+# adding Reviewed-by and then adding Fixes.
+# This also requires a hash while running tests.
+CORRECT_COMPLETE_LOG="Signed-off-by: kw <kw@kw>
+Acked-by: Michael Doe <michaeldoe@community.com>
+Reviewed-by: John Doe <johndoe@community.com>
+Fixes: <hash>
+
+Signed-off-by: kw <kw@kw>
+Acked-by: Michael Doe <michaeldoe@community.com>
+Reviewed-by: John Doe <johndoe@community.com>
+
+Signed-off-by: kw <kw@kw>
+Acked-by: Michael Doe <michaeldoe@community.com>
+Reviewed-by: John Doe <johndoe@community.com>
+
+Signed-off-by: kw <kw@kw>"
+
+function setUp()
+{
+  local -r current_path="$PWD"
+
+  cp -r tests/unit/samples/handle_trailer_patch_samples/*.patch "$SHUNIT_TMPDIR"/
+
+  cd "$SHUNIT_TMPDIR" || {
+    fail "($LINENO) It was not possible to move to temporary directory"
+    return
+  }
+  # Setup git repository for test
+  mk_fake_git
+
+  # Start repository
+  git config user.name kw
+  git config user.email kw@kw
+  mkdir fs
+  touch fs/some_file
+  git add fs/
+  git commit --quiet -s -m "fs: some_file: Fill file" -m "First"
+
+  # Simulate wrong change
+  echo 'Wrong text' > fs/some_file
+  git add fs/some_file
+  git commit --quiet -s -m "fs: some_file: Fill file" -m "First"
+
+  # Regular change
+  touch fs/new_driver
+  git add fs/new_driver
+  git commit --quiet -s -m "fs: new_driver: Add new driver" -m "Second"
+
+  # Bug fix
+  echo 'Correct text' > fs/some_file
+  git add fs/some_file
+  git commit --quiet -s -m "fs: some_file: Fix bug" -m "Third"
+
+  cd "$current_path" || {
+    fail "($LINENO) It was not possible to move back from temp directory"
+    return
+  }
+}
+
+function tearDown()
+{
+  rm -rf "$SHUNIT_TMPDIR"
+  mkdir -p "$SHUNIT_TMPDIR"
+}
+
+# This function tests 'handle_trailer_main' for
+# cases where the given argument is a patch.
+function test_handle_trailer_main_patch()
+{
+  local original_dir="$PWD"
+  cd "$SHUNIT_TMPDIR" || {
+    fail "($LINENO) It was not possible to move to temporary directory"
+    return
+  }
```

Comment:
Why not move this cd to the setup and the cd at the end of this test to the teardown?

Also: https://kworkflow.org/content/codingstyle.html#string-concatenation